agent/internal/detect.go

```diff
@@ -63,6 +69,19 @@ func (a *agent) detect() error {
 func detectCPUs() ([]device.Device, error) {
 	switch cpuInfo, err := cpu.Info(); {
 	case err != nil:
+		// Apple M1 does not report CPU frequency,
+		// and that can result in an error here
+		if errno, ok := err.(syscall.Errno); ok {
+			switch errno {
+			case syscall.ENOENT:
+				if runtime.GOARCH == "arm64" && runtime.GOOS == osDarwin {
+					return []device.Device{{ID: 0, Brand: "Apple", UUID: "AppleM1", Type: device.CPU}}, nil
```

Comment:
I think we should call it "Apple Silicon" instead of M1, because I foresee M2 next spring or next fall at the latest.